### PR TITLE
MGMT-19487: Add documentation for cluster reinstall

### PR DESCRIPTION
**Pull request by carbonin**
Document how to reinstall a cluster using the resources from an existing install.

### CODE REVIEW

File: doc/reinstall.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+You can trigger cluster installation by recreating the resources the same way they were created for the initial installation.
+
+Note:
+
+Some fields cannot be changed between install and reinstallation. These include:
+- `clusterDeployment.baseDomain`
+- `clusterDeployment.clusterName`
+- `imageClusterInstall.hostname`
+- `imageClusterInstall.nodeIP`
```
Comment (carbonin) [inline doc/reinstall.md:115]: Are these still true? Also maybe I should leave this note for when I implement the check for https://issues.redhat.com/browse/MGMT-19209?

Comment (carbonin) [reply doc/reinstall.md:115]: Looks like hostname might be relevant but node IP isn't actually used anywhere. I'll remove that one.

File: doc/reinstall.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+#### Create the resources to install the cluster
+
+You can trigger cluster installation by recreating the resources the same way they were created for the initial installation.
+
+Note:
+
+Some fields cannot be changed between install and reinstallation. These include:
+- `clusterDeployment.baseDomain`
+- `clusterDeployment.clusterName`
+- `imageClusterInstall.hostname`
```
Comment (carbonin) [inline doc/reinstall.md:114]: I assume this is true, but can you confirm @eranco74 ?

File: doc/reinstall.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+### Recreating the cluster
+
+#### Remove the existing cluster resources
+
+Delete the following resources associated with the original cluster installation:
+- `ClusterDeployment`
+- `ImageClusterInstall`
+- `BareMetalHost`
+- `DataImage`
```
Comment (sakhoury) [inline doc/reinstall.md:93]: Shouldn't the deletion of the `DataImage` object be handled by the BMO when the BMH object is deleted?

Comment (carbonin) [reply doc/reinstall.md:93]: I had some issues with this not being deleted so I added it here, but you're probably right in that _something_ is likely removing this during finalizer handling. Possibly even the ImageClusterInstall

File: doc/reinstall.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  name: ibi-test-seed-reconfiguration
+  namespace: ibi-test
+type: Opaque
+```
+
+### Recreating the cluster
+
+#### Remove the existing cluster resources
+
+Delete the following resources associated with the original cluster installation:
```
Comment (sakhoury) [inline doc/reinstall.md:89]:
```suggestion
Ensure that the following resources associated with the original cluster installation have been deleted, or delete them manually if they are still present:
```

File: doc/reinstall.md (hunk 3 above)
Comment (sakhoury) [reply doc/reinstall.md:93]: Added a suggestion to reword the deletion.

Comment (tsorya) [reply doc/reinstall.md:93]: BMO will not delete it if you delete bmh, i think they have some bug with it, so probably better just to delete it

File: doc/reinstall.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+oc get secret $CDNAME-seed-reconfiguration -oyaml > seed-reconfiguration.yaml
+```
+
+### Prepare the secrets for recreation
+
+Once the secrets are backed up some data needs to be removed to ensure they work correctly when recreated.
+
+Remove the following fields if they are present from all the previously created secret files:
+- metadata.creationTimestamp
+- metadata.labels (remove all except `image-based-installed.openshift.io/created`)
```
Comment (sakhoury) [inline doc/reinstall.md:32]: Should only the `image-based-installed.openshift.io/created` label be retained, or is it possible to keep the entire `metadata.labels` object?

**Review by sakhoury — APPROVED** [review]

**Pull request merged**